api/types/selector.go

```diff
@@ -24,4 +26,55 @@ type Selector struct {
 	// https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#api
 	// It matches with the resource labels.
 	LabelSelector string `json:"labelSelector,omitempty" yaml:"labelSelector,omitempty"`
+
+	groupRegex   *regexp.Regexp
+	versionRegex *regexp.Regexp
+	kindRegex    *regexp.Regexp
+}
+
+func anchorRegex(pattern string) string {
+	if pattern == "" {
+		return pattern
+	}
+	return "^(?:" + pattern + ")$"
+}
+
+// MatchByGvkRegex return true if gvk can be matched by regex
+// GVK in s.
+func (s *Selector) MatchByGvkRegex(gvk resid.Gvk) bool {
+	var err error
+	if s.groupRegex == nil {
+		s.groupRegex, err = regexp.Compile(anchorRegex(s.Group))
```

Comment:
- this does the compilation once (good)
- it swallows the error (bad)
- we should do the compilation immediately after reading the kustomization file and fail the build there.
- the code is now dedicated to only doing regexp matching, so even a plain string
   like `Deployment` is treated as a regexp match.  So one option is to treat Selector as a wire only type,
   and make a new struct which has only compiled regexps inside, and pass that around instead of Selector.

Reply:
The Selector is directly parsed from the YAML file. If we want to compile it immediately, we need to remember the compilation for every possible YAML parsing.

A plain string like `Deployment` will be convert to `^(?:Deployment)$` and that regex only matches the plain string `Deployment`.